BASE/src/ServerTelemetryChannel/Implementation/Transmitter.cs

```diff
@@ -169,10 +173,84 @@ internal virtual void Enqueue(Transmission transmission)
 
             if (!this.Storage.Enqueue(transmissionGetter))
             {
+                transmission.IsFlushAsyncInProgress = false;
                 TelemetryChannelEventSource.Log.TransmitterStorageSkipped(transmission.Id);
             }
         }
 
+        internal Task<bool> FlushAsync(Transmission transmission, CancellationToken cancellationToken)
+        {
+            TaskStatus taskStatus = TaskStatus.Canceled;
+            if (!cancellationToken.IsCancellationRequested)
+            {
+                transmission.IsFlushAsyncInProgress = true;
+                this.Enqueue(transmission);
```

Comment:
There is a possibility that sender and buffer and storage are all full and data is dropped inside this method. We are blind to that possibility, and won't return false to end user in this case..

Reply:
we can modify Enqueue to return its success status, and act acordingly. (in a future PR)

Reply:
This use case is handled, if sender, buffer and storage are full IsFlushAsyncInProgress is set to false in Enqueue on line 176 `transmission.IsFlushAsyncInProgress = false;`

Line 210, in method FlushAsync there is a check to handle it.